server/start.go

```diff
@@ -355,9 +355,14 @@ func setupNodeAndExecutor(
 		return nil, nil, cleanupFn, err
 	}
 
-	adapterMetrics := adapter.NopMetrics()
+	var opts []adapter.Option
 	if rollkitcfg.Instrumentation.IsPrometheusEnabled() {
-		adapterMetrics = adapter.PrometheusMetrics(config.DefaultInstrumentationConfig().Namespace, "chain_id", cmtGenDoc.ChainID)
+		m := adapter.PrometheusMetrics(config.DefaultInstrumentationConfig().Namespace, "chain_id", cmtGenDoc.ChainID)
+		opts = append(opts, adapter.WithMetrics(m))
+	}
+
+	if srvCtx.Viper.GetBool(FlagNetworkSoftConfirmation) {
```

Comment:
i think it is a easy gotcha to have some consensus failures. Cannot we detect if the module is present in the chain and enable it automatically?

Reply:
Good feedback. Not fully sure though, how to do do this during startup phase. An abci-query to the module may work but I would prefer to have an new PR covering this. I will add an issue.